File: test/Marsen.NetCore.Dojo.Tests/Classes/Joey/Tennis/States/SameState.cs
using System.Reflection.Metadata;

namespace Marsen.NetCore.Dojo.Tests.Classes.Joey.Tennis.States
{
    public class SameState : State
    {
        public override string Score()
        {
            return $"{ScoreLookup[Context.ServerPoint]} All";
        }

        // public override void ServerScore()
        // {
        //     Context.ServerPoint++;
        //     ChangeState();
        // }

        protected override void ChangeState()
        {
            State state = new NormalState();
            state.SetContext(this.Context);
            Context.ChangeState(state);
        }

        public override void ReceiverScore()
        {
            Context.ReceiverPoint++;
            ChangeState();
        }
    }
}

File: src/Marsen.TypeScript.Dojo/src/Kata/SameState.ts
import { NormalState } from "./NormalState";
import { State } from "./State";

export class SameState extends State {
    ChangeState() {
        this.Context.State = new NormalState();
    }
    Score(): string {
        return `${this.ScoreLookup.get(this.Context.ServerPoint)} All`;
    }
}
